### PR TITLE
Enhacements.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

- [x] Add github actions. 
- [x] Fix static analysis.
- [x] Fix tests php 8.0.

### CODE REVIEW

File: tests/Unit/Resolvers/ClassNameResolverTest.php [hunk 1]
```diff
@@ -21,12 +21,21 @@ public function testResolveConstructor(): void
...
+
         $this->assertCount(2, $dependencies);
+
+
         // Since reflection for built in classes does not get default values.
-        $this->assertEquals(null, $dependencies['time']->resolve($container));
+        if (version_compare(phpversion(), '8.0', '>=')) {
+            $this->assertEquals('now', $dependencies['datetime']->resolve($container));
+        } else {
+            $this->assertEquals(null, $dependencies['time']->resolve($container));
```
Comment (xepozz) [inline tests/Unit/Resolvers/ClassNameResolverTest.php:36]: `time` is modified in php 8.0?

File: src/Definitions/DynamicReference.php [hunk 2]
```diff
@@ -32,9 +32,9 @@ private function __construct($definition)
         $this->definition = Normalizer::normalize($definition);
     }
 
-    public static function to($definition): DynamicReference
+    public static function to(string $id): ReferenceInterface
```
Comment (samdark) [inline src/Definitions/DynamicReference.php:35]: That breaks what's described in the phpdoc of the class.

File: tests/Unit/Resolvers/ClassNameResolverTest.php (hunk 1 above)
Comment (samdark) [reply tests/Unit/Resolvers/ClassNameResolverTest.php:36]: Reflection is.

File: src/Definitions/DynamicReference.php (hunk 2 above)
Comment (terabytesoftw) [reply src/Definitions/DynamicReference.php:35]: Then there is something wrong the interface is clear, if I leave it as it was it does not comply with the contract.

Comment (samdark) [reply src/Definitions/DynamicReference.php:35]: Yes. There are some possible way for a fix:

1. Remove `DynamicReference`. Is it used anywhere?
2. Adjust interface by removing type.

Comment (samdark) [reply src/Definitions/DynamicReference.php:35]: This is now fixed with https://github.com/yiisoft/factory/commit/9378aae203906484e40952d1572a3ec1e1e2ab32. @terabytesoftw would you please update this pull request?

**Comment by samdark** [comment]
👍

**Pull request merged**